test/manifest.e2e-spec.ts

```
@@ -0,0 +1,382 @@
+    pubkey: pk,
+    withdrawalCredentials: fromHexString(GOOD_WC),
+    amount: 32000000000, // gwei!
+  };
+  const goodSigningRoot = computeRoot(goodDepositMessage);
+  const goodSig = sk.sign(goodSigningRoot).toBytes();
+
+  const mockedModule = {
+    nonce: 6046,
+    type: 'grouped-onchain-v1',
+    id: 1,
+    stakingModuleAddress: NOP_REGISTRY,
+    moduleFee: 10,
+    treasuryFee: 10,
+    targetShare: 10,
+    status: 1,
+    name: 'NodeOperatorRegistry',
+    lastDepositAt: block.timestamp,
+    lastDepositBlock: block.number,
+  };
+
+  const mockedMeta = {
+    blockNumber: block.number,
+    blockHash: block.hash,
+    timestamp: block.timestamp,
+  };
+
+  const mockedKey = {
+    key: toHexString(pk),
+    depositSignature: toHexString(goodSig),
+    operatorIndex: 0,
+    used: false,
+    index: 0,
+  };
+
+  jest.spyOn(keysApiService, 'getModulesList').mockImplementation(async () => ({
+    data: [mockedModule],
+    elBlockSnapshot: mockedMeta,
+  }));
+
+  jest
+    .spyOn(keysApiService, 'getUnusedModuleKeys')
+    .mockImplementation(async (_stakingModuleId: number) => ({
+      data: {
+        keys: [mockedKey],
+        module: mockedModule,
+      },
+      meta: {
+        elBlockSnapshot: mockedMeta,
+      },
+    }));
+};
+
+describe('ganache e2e tests', () => {
+  let providerService: ProviderService;
+  let walletService: WalletService;
+  let keysApiService: KeysApiService;
+  let guardianService: GuardianService;
+  let lidoService: LidoService;
+  let depositService: DepositService;
+  let blsService: BlsService;
+  let server: ReturnType<typeof makeServer>;
+
+  beforeAll(async () => {
+    server = makeServer(FORK_BLOCK, CHAIN_ID, UNLOCKED_ACCOUNTS);
+    await server.listen(GANACHE_PORT);
+  });
+
+  afterAll(async () => {
+    await server.close();
+  });
+
+  beforeAll(async () => {
+    // Prepare a signer for the unlocked Ganache account
+    if (!process.env.WALLET_PRIVATE_KEY) {
+      process.exit();
```

It may not be obvious what the problem is. Can we throw an exception with details?

---

Agreed, adding